VPLAY-9707:Fix multi-character character constant warnings

srikanthreddybijjam-comcast opened the pull request:
Reason for change: Rectified Multi char Warnings issues in L1.
Test Procedure:  Refer VPLAY-9707 for clearing warnings due to type conversion errors.
Priority: P2

Signed-off-by: Srikanth2000<srikanthreddybijjam.2000@gmail.com>

srikanthreddybijjam-comcast committed:
VPLAY-9707:Fix multi-character character constant warnings

Reason for change: Modified similar Multi char Warnings
Test Procedure:  Refer VPLAY-9707 for clearing warnings due to type conversion errors.
Priority: P2

Signed-off-by: Srikanth2000<srikanthreddybijjam.2000@gmail.com>

srikanthreddybijjam-comcast committed:
VPLAY-9707:Fix multi-character character constant warnings

Reason for change: Modified similar Multi char Warnings
Test Procedure:  Refer VPLAY-9707 for clearing warnings due to type conversion errors.
Priority: P2

Signed-off-by: Srikanth2000<srikanthreddybijjam.2000@gmail.com>

molakalapalliharipriya committed:
RDKEMW-5467 : Set PlayerState to IDLE during stop

Reason for change: EPG currently behaves badly when state is set to eSTATE_RELEASED instead of eSTATE_IDLE after stopping
Test Procedure: Tune to any linear channels, Try changing the channels and see tune was successful with out any blue screen
Risks: Medium

Signed-off-by: haripriya_molakalapalli <haripriya_molakalapalli@comcast.com>

varatharajan568 committed:
RDK-56052 Reverted PR #154 (#183)

Reason for Change: Reverted accidentally merged PR #154 from
rdkcentral/feature/RDK-56052
Test Procedure: None
Risks: Low

Signed-off-by: Varatharajan_Narayanan <Varatharajan_Narayanan@comcast.com>

dp0000 committed:
VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in … (#184)

* VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in XiOne-Foxtel
Reason for change: Updating Curldownloader instances upon calling UpdateMaxDrmSession

Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: I2b4c22d9a381f792eb08dfe8a49eecabe6f374d6
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

* VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in XiOne-Foxtel

Reason for change: Addressed review comments.
Test Procedure: Refer Ticket
Risks: Low

---------

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com
Co-authored-by: deepika <deepikasri_n@comcast.com>
Co-authored-by: Sivasubramanian Patchaiperumal <sivasubramanian.patchaiperumal@ltts.com>
Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino commented:
1. Do we really need both function and macro?
static uint32_t ConvertToDecimal(const char* text)
define CONVERT_TO_DECIMAL(Text)

2. can we define this convert glue in single place?

3. for each usage, don't need to repeat the four character 'cenc' again in comment; the use of "cenc" is self documenting

4. consider more meaningful name than generic CONVERT_TO_DECIMAL; maybe MULTICHAR_CONSTANT?

p-bond committed:
Feature/vplay 8907 guard against thread overwrite (#131)

* VPLAY-9677 Guard against fragmentCollectorThread being overwritten

Attempting to assign a thread to an existing running thread calls std::terminate()
Check joinable status and only create if the thread is not already running.

Remove boolean fragmentCollectorThreadStarted from FragmentCollector classes, as it is not needed anymore;
this is an intrinsic property of std::thread.

Testing:  L1 tests run (but investigating extending)

* VPLAY-9677 Guard against fragmentCollectorThread being overwritten
Attempting to assign a thread to an existing running thread calls std::terminate()
Check joinable status and only create if the thread is not already running.

Remove boolean fragmentCollectorThreadStarted from FragmentCollector classes, as it is not needed anymore;
this is an intrinsic property of std::thread.

Testing:  L1 tests run (but investigating extending)

* VPLAY-9677 Guard against fragmentCollectorThread being overwritten
Attempting to assign a thread to an existing running thread calls std::terminate()
Check joinable status and only create if the thread is not already running.

Remove boolean fragmentCollectorThreadStarted from FragmentCollector classes, as it is not needed anymore;
this is an intrinsic property of std::thread.

Testing:  L1 tests run (but investigating extending)

* VPLAY-9677 Guard against fragmentCollectorThread being overwritten
Attempting to assign a thread to an existing running thread calls std::terminate()
Check joinable status and only create if the thread is not already running.

Remove boolean fragmentCollectorThreadStarted from FragmentCollector classes, as it is not needed anymore;
this is an intrinsic property of std::thread.

Testing:  L1 tests run (but investigating extending)

* VPLAY-9677 Guard against fragmentCollectorThread  & TSBreaderThread being overwritten
Attempting to assign a thread to an existing running thread calls std::terminate()
Check joinable status and only create if the thread is not already running.

Remove boolean fragmentCollectorThreadStarted from FragmentCollector classes, as it is not needed anymore;
this is an intrinsic property of std::thread.

Testing:  L1 tests run (but investigating extending)

* VPLAY-9677 Guard against fragmentCollectorThread  & TSBreaderThread being overwritten
Attempting to assign a thread to an existing running thread calls std::terminate()
Check joinable status and only create if the thread is not already running.

Remove boolean fragmentCollectorThreadStarted from FragmentCollector classes, as it is not needed anymore;
this is an intrinsic property of std::thread.

Testing:  L1 tests run (but investigating extending)

* Dependency injection added for testing of thread starting
Fakes & mocks extended to cope with this
Incidental change to related tests using strict mocks

* Tidy formatting

* Formatting

* Fix l1 test

* Fix L1 test

* Review comments.

* WS

---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

srikanthreddybijjam-comcast commented:
@pstroffolino 
1. We don't have any common header files for DrmHelperTests and mp4demux to create a common MACRO in  single place.  
    Can I Define the Macro in both files.

2. Also found lot of hard coded values like follows in mp4demux.hpp.
                                case 0x68657631: // 'hev1'
				case 0x68766331: // 'hvc1'
				case 0x61766331: // 'avc1'


3. Also Can we change those by creating MACRO as 
                                               case MULTICHAR_CONSTANT("hev1"):
                                               case MULTICHAR_CONSTANT("hvc1"):
                                               case MULTICHAR_CONSTANT("avc1"):

4. Rarely getting warnings in mp4demux.hpp as well , where that MACRO can help us 
 (https://github.com/user-attachments/assets/8ce0e094-9b41-4c6b-8b4b-5e1a23afff13)

pstroffolino committed:
VPLAY-9889_2 include Content-Length in logging (#176)

* VPLAY-9889_2 include Content-Length in logging

Reason for change: allow visualization to show percent of download completed when timing out

* VPLAY-9889 add contentlength to CurlDownloader

Reason for Change: include total size (when known) in new curl-write logs

shripadbpersonal committed:
Merge pull request #194 from rdkcentral/feature/VPLAY-9931_sprint

RDKEMW-5467 : Set PlayerState to IDLE during stop

nu641001 committed:
VPLAY-9999: Progress value for latency displayed as wrong for LLD (#192)

* VPLAY-9999: Progress value for latency displayed as wrong for LLD

Reason for change:Introducing an error log to capture manifest issues
Risks: Low
Test Procedure: Test with channel 922
Priority: P1
Signed-off-by: nu641001 <nu641001@gmail.com>

* VPLAY-9999: Progress value for latency displayed as wrong

Signed-off-by: nu641001 <nu641001@gmail.com>

---------

Signed-off-by: nu641001 <nu641001@gmail.com>
Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

Commit:
VPLAY-9707:Fix multi-character character constant warning.

Reason for change:Implemented common header file.
Test Procedure: Refer VPLAY-9707 for clearing warnings.
Priority: P2

Signed-off-by: Srikanth2000srikanthreddybijjam.2000@gmail.com

Commit:
VPLAY-9707:Fix multi-character character constant warning.

Reason for change:Implemented common header file.
Test Procedure: Refer VPLAY-9707 for clearing warnings.
Priority: P2

Signed-off-by: Srikanth2000srikanthreddybijjam.2000@gmail.com

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=171) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=171) before we can accept your contribution.<br/>**10** out of **11** committers have signed the CLA.<br/><br/>:white_check_mark: srikanthreddybijjam-comcast<br/>:white_check_mark: varatharajan568<br/>:white_check_mark: dp0000<br/>:white_check_mark: pstroffolino<br/>:white_check_mark: p-bond<br/>:white_check_mark: molakalapalliharipriya<br/>:white_check_mark: nu641001<br/>:white_check_mark: rajat698<br/>:white_check_mark: jfagunde<br/>:white_check_mark: shripadbpersonal<br/>:x: Srikanth<br/><hr/>**Srikanth** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=171) it.</sub>

jfagunde committed:
VPLAY-10043 Not enable AAMP TSB for FOG urls (#165)

* VPLAY-9842 Not enable AAMP TSB for FOG urls

Reason for Change: Disable AAMP TSB if FOG is used
Test Procedure: Please see the ticket
Risks: Low

Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

* Add manifest parameter to IsFogUrl()

* Add L1 tests

* Check IsFogUrl only once

* Simplify L1 tests

* L1 test review comment

* Allow a single local TSB and extend L1 tests

* Defog the URL if FOG is not used

* L1 test comments

---------

Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>
Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

rajat698 committed:
VPLAY-10062 Build libaampKotlin library for simulator builds only (#198)

Reason for change: To enable kotlin build for simluator only
Risks: Low
Priority: P1

Signed-off-by: rajat698 <emailofrajatyadav@gmail.com>

pstroffolino committed:
VPLAY-10074 FakePlayerSecManager.cpp:170:9: warning: reference to stack memory associated with local variable 'callback' returned [-Wreturn-stack-address] (#201)

Reason for Change: avoid compilation warning with utests
Test Procedure: this warning no longer present when building l1tests
Risks: nonen

pstroffolino committed:
Merge branch 'feature/VPLAY-9707_MultiCharWarning' of https://github.com/rdkcentral/aamp into feature/VPLAY-9707_MultiCharWarning
